stan/math/prim/scal/meta/is_constant_struct.hpp

```diff
@@ -2,18 +2,27 @@
 #define STAN_MATH_PRIM_SCAL_META_IS_CONSTANT_STRUCT_HPP
 
 #include <stan/math/prim/scal/meta/is_constant.hpp>
+#include <stan/math/prim/scal/meta/conjunction.hpp>
 
 namespace stan {
-
 /**
- * Metaprogram to determine if a type has a base scalar
- * type that can be assigned to type double.
- * @tparam T Types to test
+ * Defines a public enum named value and sets it to true(1)
+ * if the provided type is constant, false(0) otherwise.
```

Comment:
The doc should explain that "constant" means a primitive type here.  

Being assignable to `double`, as used below, isn't quite right as that could be enabled by an external `operator=` definition.

For this program, I'd suggest:

```
Metaprogram defining an enum `value` whose value is `true` if the
type parameter is a primitive type and `false` otherwise.
```